### DISCUSSION TITLE
Is layering XYcharts on top of each other possible?

### DISCUSSION THREAD

**Discussion by VigneshTrantor** [airbnb/visx#1161] [Q&A] [answered]
Hey guys, 
I want to create something like this:
![image](https://user-images.githubusercontent.com/73022493/114505743-763ca480-9c4e-11eb-99ae-2ac54ed2c177.png)
i.e. 2 different charts shown on top of each other.
Is this currently possible with XYCharts ?

**Comment by williaster** [comment, marked as answer]
this should be possible, I think you'd just need to `absolute`ly position one in a container on top of the other. Two things that might be tricky are
1) matching the chart dimensions between the two if you need them to be responsive, because whichever you render inside an `absolute`ly positioned container won't have any size on its own and would render with 0 `width/height`.
2) supporting Tooltips for both (ideally you would probably want a tooltip to show data points across both charts)

Something like this might work for 1) (untested)
```tsx
import { ParentSize } from '@visx/responsive';
import { XYChart, ... } from '@visx/xychart';

// rendered inside of the container of choice, it will expand to 100% width/height
() => (
  <ParentSize>
    {({ width, height }) => (
      <>
        <XYChart width={width} height={height} {...propsAndChildren1} />
        <div style={{ position: 'absolute', top: 0, left: 0, width, height }}>
          <XYChart width={width} height={height} {...propsAndChildren2} />
        </div>
     </>
    )}
  </ParentSize>
)
```

For shared tooltips you could probably pull out the `TooltipProvider` and/or `EventProvider` as a parent of both charts (see advanced usage in the [XYChart docs](https://airbnb.io/visx/docs/xychart) for more on those)

**Reply by VigneshTrantor** [reply]
Thanks @williaster . I'll check it out.

**Reply by VigneshTrantor** [reply]
![lines](https://user-images.githubusercontent.com/73022493/114704093-cd707100-9d43-11eb-8b89-7ebae24216af.png)
I got it to work, but as you said, the tooltips were messed up. They don't work anymore. I'll try to use the `TooltipProvider` and/or `EventProvider` as suggested by you.
